src/Paprika/Store/PageManagers/MemoryMappedPageManager.cs

```diff
@@ -61,11 +61,9 @@ public unsafe MemoryMappedPageManager(long size, byte historyDepth, string dir,
             _file = new FileStream(Path, FileMode.Open, FileAccess.ReadWrite, FileShare.None, 4096,
                 PaprikaFileOptions);
         }
+        _mapped = MemoryMappedFile.CreateFromFile(_file, null, 0, MemoryMappedFileAccess.ReadWrite, HandleInheritability.None, true);
 
-        _mapped = MemoryMappedFile.CreateFromFile(_file, null, (long)size, MemoryMappedFileAccess.ReadWrite,
-            HandleInheritability.None, true);
-
-        _whole = _mapped.CreateViewAccessor();
+        _whole = _mapped.CreateViewAccessor(0, historyDepth * Page.PageSize);
```

Comment:
The `_whole` is used as both, the view to `ForceFlush` when a synchronization happens and as an accessor that provides the pointer `_ptr` for the whole file. This is why it was set to cover whole file.

Reply:
Can we introduce a new variable called _viewAccessor for accessing the memory mapped portion as a pointer. And use _whole as it is for `ForceFlush`?

Reply:
I'm not following. `ForceFlush` semantics should be that all the pages that were allocated are flushed down and then FSYNC is called. I believe we could create a one time use accessor and flush it. This might work.

Reply:
You mean, inside the ForceFlush method, we can get another accessor for _file that covers the whole file and flush it?